### ISSUE TITLE
Badge do not show javascript and php coverage

### ISSUE THREAD

**Issue by JDanP** [lemurheavy/coveralls-public#1596]
Hi! 

I use a CI with https://github.com/php-coveralls/php-coveralls#github-actions for php coverage (`app/`) and https://github.com/coverallsapp/github-action for javascript coverage (`resources/`). Everything is fine in my _**SOURCE FILES ON DEVELOP **_ section showing both coverages

![image](https://user-images.githubusercontent.com/43183212/143250713-cd848a64-e020-4cd5-acb7-eae6a7631f13.png)

The problem is the upper badge. It only shows the `app/` coverage (63%) without the `resources/` one (92%), or an average of them (77%)

![image](https://user-images.githubusercontent.com/43183212/143252884-0897310a-4fd7-4c89-82c1-ba15152d7b09.png)

I would like to refresh that badge and show the average percentage or both coverages at least. Is it possible somehow? Any suggestion is welcome.

Thanks!

**Comment by afinetooth** [comment]
Hi @JDanP. 

I discovered a couple of issues with your repo: 

First, while the default branch at Github was **develop**, on Coveralls it was showing as **master**. I re-sync'd the repo with Github (in your settings), so that coveralls picked up the latest changes. It now shows **develop**.

Second, your repo did not have an owner. We leverage the access token of the repo owner to perform some actions like sending status updates. This can sometimes happen when you add the repo to coveralls using the Coveralls API (instead of the Coveralls Web UI). In any case, I made you the repo owner and force-sent a status update for [this build](https://coveralls.io/builds/44525397).

With those two issues, I wouldn't necessarily trust anything in my build pages, so after making the above changes I took [one build](https://coveralls.io/builds/44525397) and re-ran it.

That updated the badge coverage % (to 77%) and also showed a successful status update:

![Screen Shot 2021-11-24 at 1 16 01 PM](https://user-images.githubusercontent.com/918667/143313787-9639b8bc-552a-43e1-8b84-ddd2c7d986db.png)

The only other thing I noticed that might be an issue with your build is that, you seem to indicate that your build is parallel, but that's not what the data on the build says, and there is only one (1) job in your build. Interestingly, that one job is labeled **#2**, which is unusual. (A one-job build's one job will usually be labeled **#1**.)

It may be that you mean to be sending two (2) parallel jobs there, in which case, you'll need to tell coveralls the build is parallel and should stay open until all jobs are complete.

If that's the case, let me know and I'll provide more help.

In the meantime you should be good-to-go.

**Comment by JDanP** [comment]
Thanks for all the help! So fast and so right. I had both issues in my setup! Sorry for that. Also, I think I have a third one, but I do not know where. After your built, which left coverage at 77%, there were 2 builds and they have left coverage again at 63%.

![image](https://user-images.githubusercontent.com/43183212/143328217-d36c064f-b612-437f-9c09-8801d75320d3.png)

I am not using parallel now, but I was trying to use it in some builds before I opened this issue (I restored my code without parallel again). May be the labeling problem was in those builds. On the other hand, there has to be a mistake somewhere at my workflow and to be honest I have no idea where it is. I only use one job, and this is the code in it that generate php coverage and export it:

```
      - name: Test with phpunit with coverage
        if: contains(github.ref, 'develop')
        run: |
          vendor/bin/phpunit --coverage-clover=build/logs/clover.xml
          composer global require php-coveralls/php-coveralls
          php-coveralls --coverage_clover=build/logs/clover.xml -v
        env:
          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

...and this is the code to generate javascript coverage and export it:

```
      - name: Test with jest with coverage
        if: contains(github.ref, 'develop')
        run: |
          npm run jest --coverage
        env:
          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}

      - name: Export coverage to coveralls
        if: contains(github.ref, 'develop')
        uses: coverallsapp/github-action@master
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
```


So far I don't find the mistake. Any suggestion will be so helpful. 
Thanks in advance!

**Comment by JDanP** [comment]
After the build you have run that left my coverage at 77%, my CI workflow put the coverage again at 63%. So I decided to try a build swapping the coverage exports order between the first exportation (phpunit) and the second one (javacript). It was a small fast test, but it made a difference. Now my coverage is at 76%

![image](https://user-images.githubusercontent.com/43183212/143634677-b9c5542b-a791-4b56-bc13-dd55a63c245d.png)

I will keep tracking my coverage percentage in the next features and fixes to keep an eye on this issue. Any idea of whats happening deeply will be a relief. Thanks for all the help!

**Comment by afinetooth** [comment]
Hi @JDanP.

So something strange is definitely happening  with your builds. 

The ones you created after [mine](https://coveralls.io/builds/44525397) ([Build #1501100177](https://coveralls.io/builds/44525397), the one I re-ran for you), all appeared to be doing the same thing as before and registering a lower coverage %.

I re-ran all of them and saw all the coverage %'s update themselves from ~63% to ~77%:

Excerpt of your new build history:
https://coveralls.io/repos/346156/builds

![Screen Shot 2021-11-26 at 12 49 10 PM](https://user-images.githubusercontent.com/918667/143638291-b0c3c84e-15ef-4bd6-9836-5afe35e99dde.png)

All of those builds have the same pattern, which is to say:

1. They all have ONE job (are not parallel)
2. But the one job is always marked `#2`.

What I think is happening is that, while your job is not parallel (you are not passing the `parallel: true` param), you are sending two jobs.

The first job is received by the Coveralls API and its coverage is calculated, then the second job is received and, because it has the same "build number" ([service_number](https://docs.coveralls.io/api-reference) per the [Coveralls API Reference](https://docs.coveralls.io/api-reference)), it _replaces_ the first job.

The Coveralls API treats new jobs for the same build this way, when they are not marked parallel, assuming you want to replace the coverage report for the original build. The next build is labeled `#2` (`job_name.2`) because it was the second job received for the build (in a non-parallel context).

The problem in this case, it seems, is that the coverage calculation is not re-performed using the data from the second job. 

TBH, I'm not sure why that isn't happening in your case. I suspect that calculation job is running but failing for some reason.

I can make some additional guesses about why that job would fail, but I'm much more likely to be able to help you _fix the issue_ than I am to determine why _that_ happened.

Since you're using Github Actions, can you please share the GHA build log(s) you have for one of those builds? That would help verify if there was a previous job that ran.

It will also be very helpful if you run your next build in [verbose mode](https://docs.coveralls.io/troubleshooting#put-it-in-verbose--v) so we can see some debug info in your next build log(s), specifically:

- You LCOV report (coverage report in original format)
- The JSON you're sending to the [Coveralls API](https://docs.coveralls.io/api-reference)

That JSON will show us the value you're sending for `service_number` (or `service_job_id`) for both jobs, which could validate my hypothesis.

---

I started writing this before I saw your last comment, so let me address your GHA config yaml excerpt and your next message in a follow-up message here momentarily...

**Comment by afinetooth** [comment]
@JDanP aha!

I think the cause is traceable back to your original comment (sorry):

> I use a CI with https://github.com/php-coveralls/php-coveralls#github-actions for php coverage (`app/`) and https://github.com/coverallsapp/github-action for javascript coverage (`resources/`).

Those are two (2) separate integrations, so I believe both are running, each sending a single job. But the two integrations don't know about each other, per se, and so the second job doesn't pick up on the fact that another one was created for the same build. 

Since you're using Github Actions for CI, both integrations identify their jobs with the same build number because they're part of the same GHA workflow run.

When this happens, and the job is not labeled `parallel: true`, subsequent jobs will replace previous ones.

Now, the solution here is going to be a bit more complicated, because [php-coveralls](https://github.com/php-coveralls/php-coveralls#github-actions) uses a different underlying coverage report format (XML) than the [Coveralls Github Action](https://github.com/coverallsapp/github-action) (LCOV). 

So, first, before we explore how to make your setup work with _one integration_, I'm curious if we can solve the problem by simply declaring each of your jobs as `parallel: true`, and closing the build with a third workflow step, which we refer to as the [parallel build webhook](https://docs.coveralls.io/parallel-build-webhook) (or "[parallel close webhook](https://docs.coveralls.io/parallel-build-webhook)"). 

To try that, you'll need to do the following:

1. Declare your [php-coveralls](https://github.com/php-coveralls/php-coveralls) job `parallel` by declaring the env var `COVERALLS_PARALLEL=true` [as described here](https://github.com/php-coveralls/php-coveralls#parallel-builds).
2. Declare your [Coveralls Github Action](https://github.com/coverallsapp/github-action) job `parallel` by adding the `parallel: true` param to your coveralls step [as described here](https://github.com/marketplace/actions/coveralls-github-action#complete-parallel-job-example).
3. Close your parallel build with the [parallel closed webhook](https://docs.coveralls.io/parallel-build-webhook) - In this case, why don't we leverage the [Coveralls Github Action](https://github.com/coverallsapp/github-action)'s method, which is to add a "finish" step, where you pass the `parallel-finished: true` param, [as shown in this example](https://github.com/marketplace/actions/coveralls-github-action#complete-parallel-job-example).

I'm hoping that will get your current setup working.

If you're entire workflow is in a single file, you should be fine. If not, or if you start seeing what appears to be race conditions, let me know.

**Comment by afinetooth** [comment]
P.S. @JDanP I corrected some important typos in both messages above, so you might want to re-read now for greater clarity.

**Comment by JDanP** [comment]
Hi! 

Yesterday I tried what you told me: I declared both jobs as parallel and I closed parallel build with a parallel closed webhook. It changed some numbers but they weren't the right ones. I did believe in that idea and despiste the fact that I do not have a matrix I added also a flag name parameter in both coverages exportations. It was the last missing thing! Now I am exporting both jobs correctly. Thank you so much for all the help. This was a small headache.

https://coveralls.io/builds/44824880
![image](https://user-images.githubusercontent.com/43183212/145072642-dda5a121-f132-444d-a72a-2a75c683f595.png)
